packages/components-native/src/Button/Button.tsx

```diff
@@ -119,6 +139,10 @@ export function Button({
     fullHeight && styles.fullHeight,
   ];
 
+  if (UNSAFE_style?.container) {
+    buttonStyle.push(UNSAFE_style.container);
+  }
```

Comment:
This, as well as the ternary in the other Views seems like more than we should need to do. Was the choice to do that because the tests were failing? I'm going to have to look into this a bit more

Reply:
Ok I played around with it. We can remove this `if` statement.

If I didn't have this `if` statement the tests would fail horribly. The issue is about style precedence. 
Removing this `if` statement and updating the wrapping View to be:
```
<View style={[...buttonStyle, UNSAFE_style?.container]}>
```
ensures UNSAFE is applied last and properly overrides defaults.

The rest of the Views don't need the ternary and I'll add suggestions there.

Reply:
We only have to spread `buttonStyle` because it's an array that's built up from multiple sources - the remaining views just combine a single base style with a single override